commit 99b859c7466d1745fdbf023efbb434b0e9f38e53 (refs/changes/67/2967/4)
Author: Jason King <jason.king@joyent.com>
Date:   2017-11-20T21:29:57+00:00 (1 year, 11 months ago)
    
    OS-6463 Don't allow clearing of thread name in LX on TID 1
    Reviewed by: Jerry Jelinek <jerry.jelinek@joyent.com>
    Reviewed by: Dan McDonald <danmcd@joyent.com>
    Approved by: Dan McDonald <danmcd@joyent.com>

diff --git a/usr/src/uts/common/brand/lx/syscall/lx_prctl.c b/usr/src/uts/common/brand/lx/syscall/lx_prctl.c
index 4aeb2b191e..4aecb6e9cc 100644
--- a/usr/src/uts/common/brand/lx/syscall/lx_prctl.c
+++ b/usr/src/uts/common/brand/lx/syscall/lx_prctl.c
@@ -190,6 +190,16 @@ lx_prctl(int opt, uintptr_t data)
 			return (0);
 		}
 
+		/*
+		 * We are currently choosing to not allow an empty thread
+		 * name to clear p->p_user.u_comm and p->p_user.u_psargs.
+		 * This is a slight divergence from linux behavior (which
+		 * allows this) so that we can preserve the original command.
+		 */
+		if (strlen(name) == 0) {
+			return (0);
+		}
+
 		/*
 		 * We explicitly use t->t_name here instead of name in case
 		 * a thread has come in between the above thread_setname()
